File: problems-extend/3901-4000/3986_number-of-elapsed-seconds-between-two-times/solution.java
class Solution {

    public int secondsBetweenTimes(String startTime, String endTime) {
        return seconds(endTime) - seconds(startTime);
    }

    private int seconds(String value) {
        int hours = (value.charAt(0) - '0') * 10 + (value.charAt(1) - '0');
        int minutes = (value.charAt(3) - '0') * 10 + (value.charAt(4) - '0');
        int seconds = (value.charAt(6) - '0') * 10 + (value.charAt(7) - '0');
        return hours * 3600 + minutes * 60 + seconds;
    }
}

File: problems-extend/3901-4000/3986_number-of-elapsed-seconds-between-two-times/solution.rs
impl Solution {
    pub fn seconds_between_times(start_time: String, end_time: String) -> i32 {
        let seconds = |value: &str| -> i32 {
            let hours = value[0..2].parse::<i32>().unwrap();
            let minutes = value[3..5].parse::<i32>().unwrap();
            let seconds = value[6..8].parse::<i32>().unwrap();
            hours * 3600 + minutes * 60 + seconds
        };
        seconds(&end_time) - seconds(&start_time)
    }
}
